### PR TITLE
Only make API requests in resolvers if no data exists in state

### PR DESCRIPTION
**Pull request by tofumatt**
## Summary

Addresses issue #1286.

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Comment by tofumatt** [comment]
Argh, sorry, I put this against the wrong branch. Fixing those merge conflicts now 😓

**Comment by tofumatt** [comment]
(Merge conflicts fixed.)

File: assets/js/googlesitekit/datastore/site/connection.js [hunk 1]
```diff
@@ -129,6 +131,16 @@ export const reducer = ( state, { type, payload } ) => {
 export const resolvers = {
 	*getConnection() {
 		try {
+			const registry = yield Data.commonActions.getRegistry();
+
+			const existingConnection = registry.select( STORE_NAME ).getConnection();
+
+			// If there are already accounts loaded in state, we don't want to make this request
```
Comment (aaemnnosttv) [inline assets/js/googlesitekit/datastore/site/connection.js]: Copy pasta comment 😄 
```suggestion
			// If there is already connection data loaded in state, don't make this request
```

File: assets/js/googlesitekit/datastore/site/connection.test.js [hunk 2]
```diff
@@ -122,6 +122,21 @@ describe( 'core/site connection', () => {
 				expect( connectionSelect ).toEqual( connection );
 			} );
 
+			it( 'does not make a network request if data is already in state', async () => {
+				const response = { connected: true, resettable: true };
+				registry.dispatch( STORE_NAME ).receiveConnection( response );
+
+				const connection = registry.select( STORE_NAME ).getConnection( );
```
Comment (aaemnnosttv) [inline assets/js/googlesitekit/datastore/site/connection.test.js]:
```suggestion
				const connection = registry.select( STORE_NAME ).getConnection();
```

**Pull request merged**